shell/exercism.fish

```diff
@@ -0,0 +1,49 @@
+complete -f -c exercism -n "__fish_seen_subcommand_from configure" -s a -l api -d "set API base url"
+complete -f -c exercism -n "__fish_seen_subcommand_from configure" -s s -l show -d "show settings"
+
+# Download
+complete -f -c exercism -n "__fish_use_subcommand" -a "download" -d "Downloads and saves a specified submission into the local system"
+complete -f -c exercism -n "__fish_seen_subcommand_from download" -s e -l exercise -d "the exercise slug"
+complete -f -c exercism -n "__fish_seen_subcommand_from download" -s h -l help -d "help for download"
+complete -f -c exercism -n "__fish_seen_subcommand_from download" -s T -l team -d "the team slug"
+complete -f -c exercism -n "__fish_seen_subcommand_from download" -s t -l track -d "the track ID"
+complete -f -c exercism -n "__fish_seen_subcommand_from download" -s u -l uuid -d "the solution UUID"
+
+# Help
+complete -f -c exercism -n "__fish_use_subcommand" -a "help" -d "Shows a list of commands or help for one command"
+complete -f -c exercism -n "__fish_seen_subcommand_from help" -a "configure download help open submit troubleshoot upgrade version workspace"
+
+# Open
+complete -f -c exercism -n "__fish_use_subcommand" -a "open" -d "Opens a browser to exercism.io for the specified submission."
+complete -f -c exercism -n "__fish_seen_subcommand_from open" -s h -l help -d "help for open"
+
+# Submit
+complete -f -c exercism -n "__fish_use_subcommand" -a "submit" -d "Submits a new iteration to a problem on exercism.io."
+complete -f -c exercism -n "__fish_seen_subcommand_from submit" -s h -l help -d "help for submit"
+
+# Troubleshoot
+complete -f -c exercism -n "__fish_use_subcommand" -a "troubleshoot" -d "Outputs useful debug information."
+complete -f -c exercism -n "__fish_seen_subcommand_from troubleshoot" -s f -l full-api-key -d "display full API key (censored by default)"
+complete -f -c exercism -n "__fish_seen_subcommand_from troubleshoot" -s h -l help -d "help for troubleshoot"
+
+# Upgrade
+complete -f -c exercism -n "__fish_use_subcommand" -a "upgrade" -d "Upgrades to the latest available version."
+complete -f -c exercism -n "__fish_seen_subcommand_from help" -s h -l help -d "help for help"
+
+# Version
+complete -f -c exercism -n "__fish_use_subcommand" -a "version" -d "Outputs version information."
+complete -f -c exercism -n "__fish_seen_subcommand_from version" -s l -l latest -d "check latest available version"
+complete -f -c exercism -n "__fish_seen_subcommand_from version" -s h -l help -d "help for version"
+
+# Workspace
+complete -f -c exercism -n "__fish_use_subcommand" -a "workspace" -d "Outputs the root directory for Exercism exercises."
+complete -f -c exercism -n "__fish_seen_subcommand_from workspace" -s h -l help -d "help for workspace"
+
+# Options
+complete -f -c exercism -s h -l help -d "show help"
+complete -f -c exercism -l timeout -a "(seq 0 100 100000)" -d "override default HTTP timeout"
```

Comment:
The tab completion for possible timeout values is cool. But I found the possible options a bit overwhelming - it filled my terminal. Is this a common practice you’ve seen in fish? If so, I recommend shortening the possible value range. 

Removing the range entirely has a funky behavior where if I tab after the timeout flag it tab completes to some random file. Possibly why you added the sequence?

Reply:
I can definitely shorten the range. I agree there's potentially too many options to tab through. Your analysis is spot on, I added it to prevent the files autocompleting there. Files autocompleting is fish's default behaviour, and most programs do this if there's no available options to add. I thought some range of numbers to be better than nothing, if for no other reason than to show that it's possible. I'll shorten the range.

Reply:
Is it possible in `fish` to document a value as it is in `zsh`? If so wouldn't it make sense to only have most commonly used timouts of 10, 30, 60 seconds, 5 and 10 minutes? 

In `zsh` such a completion list would roughly look like:

```
10 -- 10 seconds
30 -- 30 seconds
60 -- 1 minute
300 -- 5 minutes
600 -- 10 minutes
```

I consider such a completion much better than a plain list of values that no-one knows what they do without canceling the completion and looking them up in the documentation first.

Reply:
This is indeed possible and I have updated the completions to include this. Thanks for the suggestion, I was really unsure what to put for timeout values.

Reply:
In fish this looks like
```
➤ exercism --timeout=
…-timeout=10  (10 seconds)  …-timeout=30  (30 seconds)  …-timeout=60  (1 minute)  …-timeout=300  (5 minutes)  …-timeout=600  (10 minutes)
```